### DISCUSSION TITLE
Support for AV1 Transcoding/Delivery

### DISCUSSION THREAD

**Discussion by Baa14453** [OvenMediaLabs/OvenMediaEngine#1553] [Ideas]
I know it's been brought up a few times, but with AV1 now supported by [most browsers](https://caniuse.com/?search=av1) I could really make use of a low-bandwidth codec like AV1 for some users that have very slow internet speeds.

Please consider it for your 2024 roadmap :)

**Comment by naanlizard** [comment]
What's the state of AV1 encoders? I know OBS released a version with support, but is it only very small numbers of GPUs or CPUs that can use it?

What about playback in the browser, would LLHLS and WebRTC both still work, without increased load on the streaming server or clients?

**Reply by bchah** [reply]
GStreamer seems to have really robust support for it. On my M2 Ultra Mac Studio I can see an 'AOM AV1' encoder option in OBS but it seems locked to WHIP only for now. M3 Macs got a hardware AV1 _decoder_ but no encoder yet. The NVIDIA 40 series seems to have AV1 enc/dec on-board. And I believe Intel 11th-gen CPUs onward have hardware decode capabilities.

For devices with no hardware decode support, I think CPU-based AV1 decode is pretty similar to HEVC in terms of resource requirements.

**Reply by naanlizard** [reply]
Pretend I'm not familiar with HEVC - how's CPU encoding looking for it and AV1?

Generally, basically any CPU from the past decade can encode veryfast 1920x1080 x264 pretty easily, so it's a popular and common codec for streamers for my service (which tend not to have the latest hardware). Is that possible with AV1 or HEVC? From what I've read, performance is pretty poor, but maybe that's out of date

Thanks bchah

**Reply by bchah** [reply]
BBC has done a lot of research around these codecs and their papers are an interesting read: https://www.bbc.co.uk/rd/blog/2019-05-av1-codec-streaming-processing-hevc-vvc

Anecdotally I remember reading that for CPU-based encoding, HEVC is 8x more intensive than AVC, and 2x more intensive for decoding. This of course varies based on the different encoding parameters and the content of the video frames.

My company is an MPEG-LA licensee for AAC, AVC, HEVC and VVC so we are particularly excited to see what VVC brings to the North American market (it's already pretty big in Asia) - and if AV1 has a chance of unseating it as an open-source standard. Apple and NVIDIAs hardware implementations are promising for AV1.

**Reply by N-Storm** [reply]
> Pretend I'm not familiar with HEVC - how's CPU encoding looking for it and AV1?

It could be tweaked but still a very high load even for a modern high-end CPU. But NVidia's RTX 4080 & 4090 has hardware codec on-board which provides realtime encoding without any noticeable difference on performance. Fully supported by ffmpeg & OBS Studio for example. I'm using it literally for everything when possible - video recording, streaming, transcoding (AV1 takes 3x times less bandwidth with same perceivable quality on my opinion). And it's open source & royalty free.

So I'm bumping this idea as well. Sounds like it shouldn't be much hassle adding both NVENC hardware & ffmpeg based software backends which seems to exist already.
